github-pull-request-make: stress up to 5 tests per package

msbutler opened the pull request:
`maybe_stress` will only stress 5 tests per package, preventing Essential CI
timeout.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111273)
<!-- Reviewable:end -->

arulajmani reviewed (approved):
Thanks for sending out the revert, apologies for causing it!

msbutler committed:
github-pull-request-make: stress up to 5 tests per package

`maybe_stress` will only stress 5 tests per package, preventing Essential CI
timeout.

Epic: None

Release note: None

msbutler commented:
TFTRs!

bors p=999

msbutler commented:
bors r+ p=999
